public/javascripts/script.js

```diff
@@ -8,52 +8,51 @@ document.addEventListener('DOMContentLoaded', () => {
 
 //************START GOOGLE MAPS CODE************
 
-//Object of locations
-var locations={
-  building:{
-    center: {
-      lat: 25.766675,
-      lng: -80.196130  
-    }
-  }
-  }
 
-// Initialize and add the map
-function initMap() {
+var map, infoWindow;
+var pos = {};
 
-  var map = new google.maps.Map(
-      document.getElementById('map'), {
-          zoom: 16,
-          center: {lat: 25.766675, lng:-80.196130}
-      });
-
-        // Construct the circle for each value in citymap.
-        // Note: We scale the area of the circle based on the population.
-        for (var city in locations) {
-          // Add the circle for this city to the map.
-          var cityCircle = new google.maps.Circle({
-            strokeColor: '#FF0000',
-            strokeOpacity: 0.8,
-            strokeWeight: 2,
-            fillColor: '#FF0000',
-            fillOpacity: 0.35,
-            map: map,
-            center: {lat: 25.766675, lng:-80.196130},
-            radius: 80 //Math.sqrt(citymap[city].population) * 100
-              
-          });
-        }
-
-  // The marker, positioned at park
-  var marker = new google.maps.Marker({
-      position: building,
-      map: map
+function initMap() {
+  map = new google.maps.Map(document.getElementById('map'), {
+    center: { lat: pos.lat, lng: pos.lng },
+    zoom: 13
   });
-  //To make multiple markers
-  // var marker2 = new google.maps.Marker({
-  //     position: lake,
-  //     map: map
-  // });
-
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(function (position) {
+      pos.lat = position.coords.latitude,
+      pos.lng = position.coords.longitude,
+        infoWindow.setPosition(pos);
+        infoWindow.setContent('Location found.');
+        infoWindow.open(map);
+        map.setCenter(pos);
+        
+        var cityCircle = new google.maps.Circle({
+          strokeColor: '#FF0000',
+          strokeOpacity: 0.8,
+          strokeWeight: 2,
+          fillColor: '#FF0000',
+          fillOpacity: 0.35,
+          map: map,
+          center: { lat: pos.lat, lng: pos.lng },
+          radius: 80 //Math.sqrt(citymap[city].population) * 100
+        });
+      }, function () {
+        handleLocationError(true, infoWindow, map.getCenter());
+      });
+    } else {
+      // Browser doesn't support Geolocation
+      handleLocationError(false, infoWindow, map.getCenter());
+    }
+    
+    infoWindow = new google.maps.InfoWindow;
+    
+    function handleLocationError(browserHasGeolocation, infoWindow, pos) {
```

Comment:
Should this function not be at the top level rather than nested?  We are redefining this function every time we run the parent function.